compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/bir/BIRGen.java

```diff
@@ -1429,7 +1433,7 @@ private void createCall(BLangInvocation invocationExpr, boolean isVirtual) {
         // TODO: make vCall a new instruction to avoid package id in vCall
         if (invocationExpr.functionPointerInvocation) {
             this.env.enclBB.terminator = new BIRTerminator.FPCall(invocationExpr.pos, InstructionKind.FP_CALL,
-                    fp, args, lhsOp, invocationExpr.async, transactional, thenBB);
+                    fp, args, lhsOp, invocationExpr.async, transactional, thenBB, this.currentScope);
```

Comment:
If function call return values are assigned to a variable, would it be needed to increment the `currentScope`?

Reply:
We need to create new scopes for the local variables and increment the scope ids by 1. It has been done in https://github.com/ballerina-platform/ballerina-lang/blob/d4eb62a39664bbee06b2a624751bd453d3a5c9c2/compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/bir/BIRGen.java#L1163 and the current scopes are needed to be set for relevant instructions. So no need for the increment everywhere.